components/Tabs/src/MDCTabBarViewController.m

```diff
@@ -0,0 +1,226 @@
+
+@end
+
+@implementation MDCTabBarViewController {
+  /** For showing/hiding, Animation needs to know where it wants to end up. */
+  BOOL _tabBarWantsToBeHidden;
+}
+
+- (void)viewDidLoad {
+  [super viewDidLoad];
+  UIView *view = self.view;
+  view.clipsToBounds = YES;
+  view.autoresizingMask = UIViewAutoresizingFlexibleLeftMargin
+      | UIViewAutoresizingFlexibleWidth
+      | UIViewAutoresizingFlexibleRightMargin
+      | UIViewAutoresizingFlexibleTopMargin
+      | UIViewAutoresizingFlexibleHeight
+      | UIViewAutoresizingFlexibleBottomMargin;
+  MDCTabBar *tabBar = [[MDCTabBar alloc] initWithFrame:view.bounds];
+  tabBar.alignment = MDCTabBarAlignmentJustified;
+  tabBar.delegate = self;
+  self.tabBar = tabBar;
+  [view addSubview:tabBar];
+  [self updateTabBarItems];
+}
+
+- (void)viewDidLayoutSubviews {
+  [super viewDidLayoutSubviews];
+  [self updateLayout];
+}
+
+#pragma mark - Properties
+
+- (BOOL)tabBarHidden {
+  return self.tabBar.hidden;
+}
+
+- (void)setTabBarHidden:(BOOL)hidden {
+  [self setTabBarHidden:hidden animated:NO];
+}
+
+- (void)setTabBarHidden:(BOOL)hidden animated:(BOOL)animated {
+  if (_tabBar.hidden != hidden) {
+    if (animated) {
+      // Before entering our animation block put the view's layout in a good state.
+      [self.view layoutIfNeeded];
+    }
+    _tabBarWantsToBeHidden = hidden;
+    // Hiding the tab bar has the side effect of growing the current viewController to use that
+    // space. This does that in its layoutSubViews.
+    [self.view setNeedsLayout];
+    if (animated) {
+      if (!hidden) {
+        // If we are showing, set the state before the animation.
+        _tabBar.hidden = hidden;
+      }
+      [UIView animateWithDuration:MDCTabBarViewControllerAnimationDuration
+                       animations:^{ [self.view layoutIfNeeded]; }
+                       completion:^(BOOL finished) {
+            // If we are hiding, set the state after the animation.
+            _tabBar.hidden = hidden;
+        }];
+    } else {
+      _tabBar.hidden = hidden;
+    }
+  }
+}
+
+- (void)setViewControllers:(NSArray<UIViewController *> *)viewControllers {
+  if (![_viewControllers isEqual:viewControllers]) {
+    for (UIViewController *viewController in _viewControllers) {
+      // If the new array doesn't contain an item from the old array then remove it.
+      if (![viewControllers containsObject:viewController]) {
+       // Follow the UIKit rules for removing a child viewController.
+       [viewController willMoveToParentViewController:nil];
+        if (viewController.viewLoaded) {
+          [viewController.view removeFromSuperview];
+        }
+        [viewController removeFromParentViewController];
+      }
+    }
+    _viewControllers = [viewControllers copy];
+    [self updateTabBarItems];
+  }
+}
+
+- (void)setSelectedViewController:(nullable UIViewController *)selectedViewController {
+  if (_selectedViewController != selectedViewController) {
+#if DEBUG
```

Comment:
Consider removing the #if DEBUG here.  NSAssert is disabled in RELEASE builds by default.